fhirstore/schema/search/search_methods.py

```diff
@@ -5,48 +5,73 @@
-                    parsed_params["query_string"]["query"] = v
-                    parsed_params["query_string"]["fields"] = [m.group(1)]
-
-                if m.group(2) == "not":
-                    parsed_params["bool"]["must_not"]["match"][m.group(1)] = f"{v}"
-                if m.group(2) == "not-in":
-                    parsed_params["simple_query_string"]["query"] = f"-{v}"
-                    parsed_params["simple_query_string"]["fields"] = [m.group(1)]
-                if m.group(2) == "in":
-                    parsed_params["query_string"]["query"] = f"{v}"
-                if m.group(2) == "below":
-                    parsed_params["simple_query_string"]["query"] = f"({v})*"
-                    parsed_params["simple_query_string"]["fields"] = [m.group(1)]
-            elif n:
-                parsed_params["range"][key] = {
-                    number_prefix_matching[n.group(1)]: n.group(2)
-                }
-            elif p:
-                parsed_params["match"][key] = p.group(2)
-            elif not (m) or not (n):
-                parsed_params["match"][key] = v
-
-    return parsed_params
+
+    element_query = defaultdict(lambda: defaultdict(dict))
+
+    numeric_modif = re.search(r"^(gt|lt|ge|le)([0-9].*)$", f"{value}")
+    eq_modif = re.search(r"^(eq)([0-9].*)$", f"{value}")
+    str_modif = re.search(
+        r"^(.*):(contains|exact|above|below|not|in|not-in|of-type)$", key
+    )
+    if str_modif:
+        if str_modif.group(2) == "contains":
+            element_query["query_string"]["query"] = f"*{value}*"
+            element_query["query_string"]["default_field"] = str_modif.group(1)
+
+        if str_modif.group(2) == "exact":
+            element_query["query_string"]["query"] = value
+            element_query["query_string"]["fields"] = [str_modif.group(1)]
+
+        if str_modif.group(2) == "not":
+            element_query["bool"]["must_not"]["match"][str_modif.group(1)] = f"{value}"
+
+        if str_modif.group(2) == "not-in":
+            element_query["simple_query_string"]["query"] = f"-{value}"
+            element_query["simple_query_string"]["fields"] = [str_modif.group(1)]
+
+        if str_modif.group(2) == "in":
+            element_query["query_string"]["query"] = f"{value}"
+
+        if str_modif.group(2) == "below":
+            element_query["simple_query_string"]["query"] = f"({value})*"
+            element_query["simple_query_string"]["fields"] = [str_modif.group(1)]
+
+    elif numeric_modif:
+        element_query["range"][key] = {
+            number_prefix_matching[numeric_modif.group(1)]: numeric_modif.group(2)
+        }
+    elif eq_modif:
+        element_query["match"][key] = eq_modif.group(2)
+    else:
+        element_query["match"][key] = value
+
+    return element_query
+
+
+def build_simple_query(sub_param):
+    """Accepts a dictionary of length 1
+    """
+    content = []
+    sub_query = {}
+    if bool(sub_param.get("multiple")):
```

Comment:
why do you use bool() here ? if .get("...") fails, it returns None which has a similar effect as `False`

Reply:
You're right, I fixed it!